recipes/libxml2/all/conanfile.py

```diff
@@ -97,52 +97,91 @@ def requirements(self):
         if self.options.icu:
             self.requires("icu/71.1")
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
     def build_requirements(self):
-        if not (self._is_msvc or self._is_mingw_windows):
+        if not (is_msvc(self) or self._is_mingw_windows):
             if self.options.zlib or self.options.lzma or self.options.icu:
-                self.build_requires("pkgconf/1.9.3")
-            if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-                self.build_requires("msys2/cci.latest")
+                self.tool_requires("pkgconf/1.9.3")
+            if self._settings_build.os == "Windows":
+                self.win_bash = True
+                if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                    self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        # can't use strip_root here because if fails since 2.9.10 with:
-        # KeyError: "linkname 'libxml2-2.9.1x/test/relaxng/ambig_name-class.xml' not found"
-        get(self, **self.conan_data["sources"][self.version])
-        rename(self, "libxml2-{}".format(self.version), self._source_subfolder)
-
-    @contextmanager
-    def _msvc_build_environment(self):
-        with chdir(self, os.path.join(self._source_subfolder, 'win32')):
-            with tools.vcvars(self.settings):
-                with tools.environment_append(VisualStudioBuildEnvironment(self).vars):
-                    yield
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        if is_msvc(self):
+            tc = VCVars(self)
+            tc.generate()
+            env = Environment()
+            for var, value in VisualStudioBuildEnvironment(self).vars.items():
+                env.define(var, value)
```

Comment:
```suggestion
```

I've tested already while fixing libmp3lame recipe, and it doesn't work. The workaround with injection of cflags from AutotoolsToolchain in CL en var is sufficient.